cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/TopicContentsService.java

```diff
@@ -72,7 +79,8 @@ public Map<Long, String> readEvents(
     Set<TopicPartition> topicPartitionsSet = consumer.assignment();
 
     Set<TopicPartition> partitionsAssignment = new HashSet<>();
-    if (offsetPosition.equals(CUSTOM_OFFSET_SELECTION)) {
+    if (offsetPosition.equals(CUSTOM_OFFSET_SELECTION)
+        || offsetPosition.equals(RANGE_OFFSET_SELECTION)) {
```

Comment:
In case of RANGE_OFFSET_SELECTION, we should check if start and end offsets are present.

Reply:
From my testing if the url pattern is not respected then there is a 403, thats why I had to change [ClusterApiService.java](https://github.com/Aiven-Open/klaw/pull/2551/files#diff-d69349b4ecda61b4387e28e545e62bb6b0d3be8d3bf8369be0bd285b2fe8838bR284). So it is impossible that these values will not be present at all. Therefore I made changes to ensure that they are not negative.